src/release.rs

```diff
@@ -279,6 +279,90 @@ pub fn convert_to_install_only<W: Write>(reader: impl BufRead, writer: W) -> Res
+    let mut tar_in = tar::Archive::new(dctx);
+    let temp_dir = tempfile::tempdir()?;
+    tar_in.unpack(temp_dir.path())?;
+
+    // On macOS, strip the `.dylib` files.
+    if cfg!(target_os = "macos") {
+        for file in std::fs::read_dir(temp_dir.path().join("python/lib"))? {
+            let file = file?;
+
+            if file.path().extension().is_some_and(|ext| ext == "dylib") {
+                let size_before = file.metadata()?.len();
+
+                let output = std::process::Command::new("strip")
+                    .arg("-S")
+                    .arg("-x")
+                    .arg(file.path())
+                    .output()?;
+                if !output.status.success() {
+                    return Err(anyhow!("strip failed: {:?}", output));
+                }
+
+                let size_after = file.metadata()?.len();
+                println!(
+                    "stripped {} from {size_before} to {size_after} bytes",
+                    file.path().strip_prefix(temp_dir.path()).unwrap().display(),
+                );
+            }
+        }
+    }
+
+    // On macOS, strip the `.dylib` files.
+    if cfg!(target_os = "linux") {
+        for file in std::fs::read_dir(temp_dir.path().join("python/lib"))? {
+            let file = file?;
+
+            if file.path().extension().is_some_and(|ext| ext == "so") {
+                let size_before = file.metadata()?.len();
+
+                let output = std::process::Command::new("strip")
+                    .arg("--strip-unneeded")
+                    .arg(file.path())
+                    .output()?;
+                if !output.status.success() {
+                    return Err(anyhow!("strip failed: {:?}", output));
+                }
+
+                let size_after = file.metadata()?.len();
+                println!(
+                    "stripped {} from {size_before} to {size_after} bytes",
+                    file.path().strip_prefix(temp_dir.path()).unwrap().display(),
+                );
+            }
+        }
+    }
+
+    // On Windows, remove `.pdb` files.
+    if cfg!(target_os = "windows") {
+        for file in std::fs::read_dir(temp_dir.path().join("python"))? {
+            let file = file?;
+            if file.path().extension().is_some_and(|ext| ext == "pdb") {
+                std::fs::remove_file(file.path())?;
+            }
+        }
+
+        for file in std::fs::read_dir(temp_dir.path().join("python/DLLs"))? {
+            let file = file?;
+            if file.path().extension().is_some_and(|ext| ext == "pdb") {
+                std::fs::remove_file(file.path())?;
+            }
+        }
+    }
+
+    // Tar the directory back up and write it to the writer.
+    let writer = flate2::write::GzEncoder::new(writer, flate2::Compression::default());
+    let mut builder = tar::Builder::new(writer);
+    builder.append_dir_all("python", temp_dir.path())?;
+    Ok(builder.into_inner()?.finish()?)
```

Comment:
This almost certainly loses the determinism that the original `.tar.gz` are produced with.

See the Rust code in `convert_to_install_only()` for how to make tar creation deterministic.

Instead of extracting the entire archive to a temporary directory and re-tarring it later, I would iterate over tar entries, do an in-memory extract of the file content, and then write to a temporary file to run `llvm-strip` if and only if necessary. This will allow you to preserve the order and metadata of the source tar archive and preserve determinism. It also drastically reduces I/O to create this release artifacts.